darshan-runtime/lib/darshan-core.c

```diff
@@ -394,12 +407,37 @@ void darshan_core_initialize(int argc, char **argv)
         darshan_core_fprintf(stderr, "darshan:init\t%d\t%f\n", nprocs, init_time);
     }
 
+
+#ifdef HAVE_LDMS
+        /*TODO: Create environment variable to re-connect to ldms every x seconds
+        if(getenv("DARSHAN_LDMS_REINIT"))
+            dC.env_ldms_reinit = getenv("DARSHAN_LDMS_REINIT");
+        else
+            dC.env_ldms_reinit = "1";
+        */
+        /* Set meta data for LDMS message sending */
+        (void)gethostname(dC.hname, sizeof(dC.hname));
+        dC.jobid = (int64_t)jobid;
+        dC.uid = getuid();
+        dC.exename = argv[0];
+
+        /* Pull executable name from darshans variable if no arguemments are given. */
+        if (argc==0)
+        {
+            char buff[DARSHAN_EXE_LEN];
+            int len = readlink("/proc/self/exe", buff, sizeof(buff)-1);
+            buff[len] = '\0';
+            dC.exename = buff;
```

Comment:
Most of this info is already captured by Darshan and I think we could just reuse it instead of redoing the work here.

- `jobid`: you are already using the value captured by Darshan, so all good
- `uid`: you can just use the value in the Darshan job structure from above: `init_core->log_job_p->uid`
- `exename`: this one is a little trickier, but I think you can actually just use `init_core->log_exemnt_p`, which is a null-terminated string of the command line and it's arguments.

Additionally, I notice that there is already a codepath in the LDMS initialize routine that calls `gethostname()` (see `setup_connection()`, line 70). Could we consolidate down to one call there? Actually, looking closer, it looks like you use the environment variable `DARSHAN_LDMS_HOST` to override this in all cases, so is the call even needed at all?